### PR TITLE
ROS#167 - Requirements coverage

### PR DESCRIPTION
**Pull request by Glamhoth**

### CODE REVIEW

File: testbins/test-boolean-condition-interrupt/expanded.rs [hunk 1]
```diff
@@ -0,0 +1,91 @@
+#![feature(prelude_import)]
```
Comment (SteelPh0enix) [inline testbins/test-boolean-condition-interrupt/expanded.rs]: wtf happened here

File: src/boolean_condition.rs [hunk 2]
```diff
@@ -101,3 +101,5 @@ impl DataProvider<bool> for BooleanCondition {
         false
     }
 }
+
+unsafe impl Sync for BooleanCondition {}
```
[the remark this answers is not in the document]
Comment (Glamhoth) [reply src/boolean_condition.rs:108]: Added

File: src/event.rs [hunk 3]
```diff
@@ -79,3 +79,5 @@ impl PartialEq for Event {
         self.id.eq(&other.id)
     }
 }
+
+unsafe impl Sync for Event {}
```
[the remark this answers is not in the document]
Comment (Glamhoth) [reply src/event.rs:85]: Added

File: src/message_queue.rs [hunk 4]
```diff
@@ -120,3 +125,5 @@ mod tests {
         assert_eq!(queue100u64_size, stub_size);
     }
 }
+
+unsafe impl<T, const N: usize> Sync for MessageQueue<T, N> {}
```
[the remark this answers is not in the document]
Comment (Glamhoth) [reply src/message_queue.rs]: Added

File: testbins/test-boolean-condition-interrupt/expanded.rs (hunk 1 above)
Comment (Glamhoth) [reply testbins/test-boolean-condition-interrupt/expanded.rs]: :D

File: src/message_queue.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (Glamhoth) [reply src/message_queue.rs]: I was checking something with `cargo expand` and then forgot to remove that file (which then was copy-pasted when I created another tests)

**Review by SteelPh0enix — APPROVED** [review]

**Pull request merged**